### DISCUSSION TITLE
@Nullable and @NotNull warnings

### DISCUSSION THREAD

**Discussion by cardosojc** [JetBrains/Qodana#63] [Q&A] [answered]
In my repo i'm using lombok and while running qodana github scan action i get 
![image](https://user-images.githubusercontent.com/8087952/151156087-ea2c7280-4935-489b-8158-d28998bb59da.png)

I know that we can configure this setting in an inspection profile and use it with qodana but i would like to stick with `qodana.recommended`

Is there any configuration we could apply in `qodana.yml` to overcome this ?

Note: when i run qodana locally i don't have this warning, probably because of my local inspection profile.

**Comment by brichbash** [comment, marked as answer]
Hi,

Unfortunately, you can't configure inspection options via `qodana.yaml` yet, but we're working on it.
The settings for used annotations are stored in `.idea/misc.xml` and that is more likely to be the reason why you don't have the warning when running locally. Could you please try creating `.idea/misc.xml` (you can take it from your local instance of the project, it should have `<component name="NullableNotNullManager">` component) a step before the analysis?

**Comment by cardosojc** [comment]
I think for now I can live with an exclude rule and remove it when
inspection options are implemented.

brichbash ***@***.***> escreveu no dia quarta, 26/01/2022
à(s) 12:06:

> Hi,
>
> Unfortunately, you can't configure inspection options via qodana.yaml
> yet, but we're working on it.
> The settings for used annotations are stored in .idea/misc.xml and that
> is more likely to be the reason why you don't have the warning when running
> locally. Could you please try creating .idea/misc.xml (you can take it
> from your local instance of the project, it should have <component
> name="NullableNotNullManager"> component) a step before the analysis?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/JetBrains/Qodana/discussions/63#discussioncomment-2050352>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AB5WTEGT6TUXEPDIQIKGKWLUX7PUXANCNFSM5M2ZPSZA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
José Cardoso

**Comment by mridang** [comment]
Is this still the case?

**Reply by brichbash** [reply]
Currently, yes.

Although now you can configure inspection options using `qodana.yaml`, the annotations config isn't stored inside the profile, but in a separate `misc.xml` file, so unfortunately, it still cannot be configured using `qodana.yaml`
